tests/interlocked_int/interlocked_int.c

```diff
@@ -0,0 +1,855 @@
+//Copyright(c) Microsoft.All rights reserved.
+//Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#ifdef __cplusplus
+#include <cstdlib>
+#include <cstddef>
+#include <cstdint>
+#else
+#include <stdlib.h>
+#include <stddef.h>
+#include <stdint.h>
+#endif
+
+
+#include "testrunnerswitcher.h"
+
+static TEST_MUTEX_HANDLE g_testByTest;
+
+
+
+#include "azure_c_util/interlocked.h"
+
+
+BEGIN_TEST_SUITE(interlocked_int)
+
+
+TEST_SUITE_INITIALIZE(a)
+{
+    g_testByTest = TEST_MUTEX_CREATE();
+    ASSERT_IS_NOT_NULL(g_testByTest);
+
+    
+}
+
+TEST_SUITE_CLEANUP(b)
+{
+
+    TEST_MUTEX_DESTROY(g_testByTest);
+}
+
+TEST_FUNCTION_INITIALIZE(c)
+{
+    if (TEST_MUTEX_ACQUIRE(g_testByTest))
+    {
+        ASSERT_FAIL("our mutex is ABANDONED. Failure in test framework");
+    }
+
+}
+
+TEST_FUNCTION_CLEANUP(d)
+{
+    TEST_MUTEX_RELEASE(g_testByTest);
+}
+
+/*
+Tests_SRS_INTERLOCKED_43_001 [ interlocked_add shall atomically add *addend with value and store the result in *addend.]
+Tests_SRS_INTERLOCKED_43_032: [interlocked_add shall return the result of the addition.]
+*/
+
+TEST_FUNCTION(interlocked_add_does_addition)
+{
+    ///arrange
+    volatile int32_t addend = INT32_MAX;
+    int32_t value = INT32_MIN;
+
+    ///act
+    int32_t return_val = interlocked_add(&addend, value);
+
+    ///assert
+    ASSERT_ARE_EQUAL(int32_t, -1, addend);
```

Comment:
>addend [](start = 34, length = 6)

Frankly, seeing as "added" is in interlocked variable, it should be read with an interlocked operation.

here and everywhere.